Libraries/PeriphDrivers/Source/UART/uart_reva.c

```diff
@@ -617,12 +684,63 @@ unsigned int MXC_UART_RevA_WriteTXFIFO(mxc_uart_reva_regs_t *uart, unsigned char
     return written;
 }
 
+int MXC_UART_RevA_SetAutoDMAHandlers(mxc_uart_reva_regs_t *uart, bool enable)
+{
+    int n = MXC_UART_GET_IDX((mxc_uart_regs_t *)uart);
+    if (n < 0)
+        return E_BAD_PARAM;
+
+    states[n].auto_dma_handlers = enable;
+
+    return E_NO_ERROR;
+}
+
+int MXC_UART_RevA_SetTXDMAChannel(mxc_uart_reva_regs_t *uart, unsigned int channel)
+{
+    int n = MXC_UART_GET_IDX((mxc_uart_regs_t *)uart);
+    if (n < 0)
+        return E_BAD_PARAM;
+
+    states[n].channelTx = channel;
+
+    return E_NO_ERROR;
+}
+
+int MXC_UART_RevA_GetTXDMAChannel(mxc_uart_reva_regs_t *uart)
+{
+    int n = MXC_UART_GET_IDX((mxc_uart_regs_t *)uart);
+    if (n < 0)
+        return E_BAD_PARAM;
+
+    return states[n].channelTx;
+}
+
+int MXC_UART_RevA_SetRXDMAChannel(mxc_uart_reva_regs_t *uart, unsigned int channel)
+{
+    int n = MXC_UART_GET_IDX((mxc_uart_regs_t *)uart);
+    if (n < 0)
+        return E_BAD_PARAM;
+
+    states[n].channelRx = channel;
+
+    return E_NO_ERROR;
+}
+
+int MXC_UART_RevA_GetRXDMAChannel(mxc_uart_reva_regs_t *uart)
+{
+    int n = MXC_UART_GET_IDX((mxc_uart_regs_t *)uart);
+    if (n < 0)
+        return E_BAD_PARAM;
+
+    return states[n].channelRx;
+}
+
 unsigned int MXC_UART_RevA_WriteTXFIFODMA(mxc_uart_reva_regs_t *uart, mxc_dma_regs_t *dma,
                                           unsigned char *bytes, unsigned int len,
                                           mxc_uart_dma_complete_cb_t callback,
                                           mxc_dma_config_t config)
 {
-    uint8_t channel;
+    uint8_t channel = -1;
```

Comment:
Assigning -1 to unsigned integer.

Reply:
https://github.com/Analog-Devices-MSDK/msdk/pull/763/commits/dc04d78fd1b005b7da86f3e75e341b9a0868417a